src/app.ts

```diff
@@ -1,162 +1,181 @@
-
-  addRestaurant = (event: Event) => {
-    event.preventDefault();
-
+  addRestaurant = ({ detail }: CustomEvent) => {
     try {
-      const restaurant = this.createRestaurant(event);
+      const restaurant = this.createRestaurant(detail);
       this.#restaurants.push(restaurant);
     } catch (e) {
       const error = e as Error;
-      alert(error.message);
+      render.message('error', 'top', error.message);
       return;
     }
 
-    this.$modal.close();
-    this.updateRestaurants();
+    render.message('success', 'bottom', ALERT_MESSAGE.createRestaurant);
+    render.closeRegisterRestaurantModal();
+    this.updateRestaurantsList();
   };
 
-  createRestaurant = (event: Event) => {
-    const restaurantProps = Object.fromEntries([
-      ...new FormData(event.target as HTMLFormElement).entries(),
-    ]);
-
+  createRestaurant = ({
+    category,
+    name,
+    distanceByMinutes,
+    description,
+    referenceUrl,
+  }: RestaurantProps) => {
     return new Restaurant({
-      category: String(restaurantProps.category),
-      name: String(restaurantProps.name),
-      distanceByMinutes: Number(restaurantProps.distanceByMinutes),
-      description: String(restaurantProps.description),
-      referenceUrl: String(restaurantProps.referenceUrl),
+      category,
+      name,
+      distanceByMinutes,
+      description,
+      referenceUrl,
+      existRestaurantsName: this.#restaurants.map((restaurant) => restaurant.getName()),
     });
   };
 
-  initSelect() {
-    this.$restaurantFilterSelect.setOptions(DEFAULT_FILTER_OPTIONS);
-    this.$restaurantSortSelect.setOptions(DEFAULT_SORT_OPTIONS);
-  }
+  openRestaurantDetailModal = ({ detail }: CustomEvent) => {
+    const { name } = detail;
 
-  initModalSelect() {
-    this.$restaurantModalCategory.setOptions(DEFAULT_MODAL_CATEGORY_OPTIONS);
-    this.$restaurantModalDistance.setOptions(DEFAULT_MODAL_DISTANCE_OPTIONS);
-  }
+    const restaurant = this.#restaurants.filter((_restaurant) => _restaurant.getName() === name)[0];
+
+    render.openRestaurantDetailModal(restaurant);
+  };
+
+  initLoad = () => {
+    render.init();
+
+    const restaurants: Restaurant[] = getRestaurants();
+
+    if (!!restaurants.length) {
+      this.#restaurants = restaurants.map((restaurant: Restaurant) =>
+        Object.setPrototypeOf(restaurant, Restaurant.prototype),
+      );
+    }
+
+    this.updateRestaurantsList();
+  };
+
+  deleteRestaurant = ({ detail }: CustomEvent) => {
+    this.#restaurants = this.#restaurants.filter(
+      (restaurant) => restaurant.getName() !== detail.name,
+    );
+
+    render.message('success', 'bottom', ALERT_MESSAGE.removeRestaurant);
+    this.updateRestaurantsList();
+    render.closeRestaurantDetailModal();
+  };
+
+  toggleRestaurantFavorite = ({ detail }: CustomEvent) => {
+    const { restaurantName } = detail;
+
+    this.#restaurants.forEach((restaurant) => {
+      if (restaurant.getName() === restaurantName) restaurant.toggleFavorite();
+    });
```

Comment:
`restaurantName`과 같은 이름의 restaurant를 찾아서 변경해주는거라면 `forEach()`보다는 `find()`를 사용하는 게 더 자연스러울거 같아요
```suggestion
    this.#restaurants
        .find((restaurant) => (restaurant.getName() === restaurantName))
        .toggleFavorite()
    });
```